cmd/lifecycle/restorer.go

```diff
@@ -90,10 +79,52 @@ func (r *restoreCmd) Privileges() error {
 }
 
 func (r *restoreCmd) Exec() error {
-	if r.supportsBuildImageExtension() {
-		if err := r.pullBuilderImageIfNeeded(); err != nil {
-			return cmd.FailErr(err, "read builder image")
+	var analyzedMD platform.AnalyzedMetadata
+	if _, err := toml.DecodeFile(r.AnalyzedPath, &analyzedMD); err == nil {
+		if r.supportsBuildImageExtension() {
+			cmd.DefaultLogger.Debugf("Pulling builder image metadata...")
+			_, digest, err := r.pullSparse(r.BuildImageRef)
+			if err != nil {
+				return cmd.FailErr(err, "read builder image")
+			}
+			analyzedMD.BuildImage = &platform.ImageIdentifier{Reference: digest.String()}
 		}
+		if r.supportsRunImageExtension() && needsPulling(analyzedMD.RunImage) {
+			cmd.DefaultLogger.Debugf("Pulling run image metadata...")
+			runImage, digest, err := r.pullSparse(analyzedMD.RunImage.Reference)
+			if err != nil {
+				return cmd.FailErr(err, "reading run image")
+			}
+			targetData, err := platform.ReadTargetData(runImage)
+			if err != nil {
+				return cmd.FailErr(err, "reading target data from run image")
+			}
+			analyzedMD.RunImage = &platform.RunImage{
+				Reference:      digest.String(),
+				Extend:         true,
+				TargetMetadata: &targetData,
+			}
+		} else if needsUpdating(analyzedMD.RunImage) {
+			cmd.DefaultLogger.Debugf("Updating analyzed metadata...")
+			runImage, digest, err := newRemoteImage(analyzedMD.RunImage.Reference, r.keychain)
+			if err != nil {
+				return cmd.FailErr(err, "reading run image")
+			}
+			targetData, err := platform.ReadTargetData(runImage)
+			if err != nil {
+				return cmd.FailErr(err, "reading target data from run image")
+			}
```

Comment:
I overlooked the daemon/layout case here. We need https://github.com/buildpacks/imgutil/pull/184 to land, then we could use that instead to get target data for all export formats.